odc/stac/eo3/_eo3converter.py

```diff
@@ -20,7 +20,7 @@
 except ImportError:
     from datacube.index.abstract import default_metadata_type_docs  # type: ignore
 
-from datacube.model import Dataset, DatasetType, metadata_from_doc
+from datacube.model import Dataset, Product, metadata_from_doc
```

Comment:
this change implies raising a lower bound on `datacube` dependency, please add that to 

https://github.com/opendatacube/odc-stac/blob/8b2e5a01712335a4930fd3d434c0d62f760f0b33/setup.cfg#L61